services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/ApplicationConfig.java

```diff
@@ -177,6 +184,17 @@ public MongoClient mongoClient() {
         return MongoClient.createShared(vertx(), mongoDbConfigProperties().getMongoClientConfig());
     }
 
+    /**
+     * Gets a {@link MongoDbCallExecutor} instance containing helper methods for mongodb interaction.
+     *
+     * @return An instance of the helper class {@link MongoDbCallExecutor}.
+     */
+    @Bean
+    @Scope("prototype")
+    public MongoDbCallExecutor mongoDBCallExecutor() {
+        return new MongoDbCallExecutor(vertx(), mongoClient());
+    }
+
```

Comment:
please follow the approach taken for the RegistrationService and create a (private) instance in the TenantService implementation's constructor.

Reply:
removed